docs/concepts/firmware-update/rfd.md

```diff
@@ -0,0 +1,402 @@
+group of servers and available firmware versions for a particular manufacturer-model:
+
+- ServerFirmware
+- ServerFirmwareGroup
+- DiscoveredFirmware
+
+All the following CRs MUST be cluster-scoped.
+The firmware versions defined for concrete `ServerFirmware` object MUST take precedence over those provided by
+corresponding `ServerFirmwareGroup` object.
+
+#### ServerFirmware
+
+`ServerFirmware` CR represents the desired state of concrete hardware server.
+The `.spec` of this type references the `Server` object, reflects its `.status.bios` field and contains the list of
+firmwares desired to be installed.
+The `.status` of this type contains information about the BIOS/firmware versions which are actually installed on the
+server.
+Aside from that `.spec` contains the scan threshold and the `.status` contains last scan operation timestamp.
+These two fields allow calculating whether the scanning for installed firmware is required or not.
+The `ServerFirmware` object SHOULD be created along with corresponding `Server` object and MUST be unique across
+the cluster.
+
+```yaml
+apiVersion: metal.ironcore.dev/v1alpha1
+kind: ServerFirmware
+metadata:
+  name: foo
+spec:
+  scanThreshold: 30m
+  serverRef:
+    name: foo
+  bios:
+    version: 1.0.0
+  firmwares:
+    - name: ssd
+      manufacturer: ACME Corp.
+      version: 1.0.0
+    - name: nic
+      manufacturer: Intel
+      version: 2.0.0
+status:
+  lastScanTime: 01-01-2001 01:00:00
+  bios:
+    version: 1.0.0
+  firmwares:
+    - name: ssd
+      manufacturer: ACME Corp.
+      version: 1.0.0
+    - name: nic
+      manufacturer: Intel
+      version: 2.0.0
+```
+
+#### ServerFirmwareGroup
+
+`ServerFirmwareGroup` CR represents the groups of hardware servers and their desired firmware versions.
+The group of servers MUST be manufacturer- and model-specific to ensure that defined firmware will be applicable.
+The `.spec` of this type contains built-in label selector and the list of firmwares desired to be installed.
+The `.status` of this type contains the information about number of servers within the defined group which are in
+desired state.
+The group MUST have a unique selector and selectors in different `ServerFirmwareGroup` objects MUST NOT intersect.
+
+```yaml
+apiVersion: metal.ironcore.dev/v1alpha1
+kind: ServerFirmwareGroup
+metadata:
+  name: bar-group
+spec:
+  manufacturer: Lenovo
+  model: 7x21
+  serverSelector:
+    matchLabels:
+      env: prod
+  firmwares:
+    - name: ssd
+      manufacturer: ACME Corp.
+      version: 1.0.0
+    - name: nic
+      manufacturer: Intel
+      version: 2.0.0
+status:
+  serversInGroup: 4
+  updatesApplied: 3
+  updatesNotApplied: 1
+```
+
+#### DiscoveredFirmware
+
+`DiscoveredFirmware` CR represents discovered firmware versions for a specific manufacturer-model.
+The `.spec` of this type contains information about manufacture, concrete model,the desired number of versions to
```

Comment:
The `.spec` of this type contains information about manufacture, concrete model, the desired number of versions to

Reply:
Yeah, hard wrapping forced by IDE. Hope now it's better.